roles/modify_container_image/tasks/yum_update_buildah.yml

```diff
@@ -107,7 +110,7 @@
       command: >
         buildah run
           --volume {{ yum_update.path }}:/tmp/yum_update.sh:z
-          --volume {{ yum_repos_dir_path }}:/etc/yum.repos.d
+          --volume {{ yum_repos_dir_path }}:/etc/yum.repos.d:z
```

Comment:
I vaguely remember we intentionally removed :z to avoid chcon over system /etc/yum.repos.d in case that directory is used.

Reply:
Okay, I'll add a test so that if `yum_repos_dir_path` is `/etc/yum.repos.d` the "z" isn't added.

Reply:
For the next itteration of this PR I'm going to leave the ':z' in place as I can't confirm that recollection:

```
[tony@thor tcib]$ git -C /home/tony/src/github.com/openstack-k8s-operators/tcib log origin/main -- roles/modify_container_image/tasks/yum_update_buildah.yml
commit 11a1ff48bfc0fdcc8a8a984d42507f5215ccd5a5
Author: Chandan Kumar <raukadah@gmail.com>
Date:   Thu Feb 23 11:12:09 2023 +0530

    Import modify image role in tcib
    
    It will be used with openstack tcib container image hotfix
    command.
    
    Signed-off-by: Chandan Kumar <raukadah@gmail.com>
```
and the "source" project has a change very similar to mine:
```
[tony@thor tcib]$ git -C /tmp/ansible-role-tripleo-modify-image log -1 origin/master -- tasks/yum_update_buildah.yml
commit b6eedb6fb194ec32b92866257131714e02bae741 (HEAD -> master, origin/master, origin/HEAD)
Author: Cédric Jeanneret <cjeanner@redhat.com>
Date:   Wed Jan 18 16:26:01 2023 +0100

    Ensure update_yum.sh has correct SELinux labels
    
    On recent, enforcing systems, the update_yum.sh can't be executed,
    because SELinux prevents container_t to open user_tmp_t:
    
    type=AVC msg=audit(1674049913.380:22858): avc:  denied  { open } for
        pid=70472 comm="bash" path="/tmp/yum_update.sh" dev="vda4" ino=218200014
        scontext=system_u:system_r:container_t:s0:c65,c705
        tcontext=unconfined_u:object_r:user_tmp_t:s0
        tclass=file permissive=0
    
    This patch ensures it gets properly relabelled when bind-mounted during
    the image build. Using the "z" will also ensure it's still usable even
    when running multiple builds at the same time.
    
    Change-Id: I4085865965f48c9fa6a88cde7010a51cd8c653d8
```

Reply:
No this change fixes the selinux error about the yum_update.sh script. I double-checked the current master of ansible-role-tripleo-modify-image it does not have z to mount repository directory.

https://github.com/openstack/ansible-role-tripleo-modify-image/blob/master/tasks/yum_update_buildah.yml#L110

I'd suggest we copy `/etc/yum.repos.d` to a different path (eg `/tmp/yum.repos.d`) and mount that directory if we really need :z flag added.